### PR TITLE
Add Fork/Join design pattern (#3227)

### LINKED ISSUE #3227: Implement Fork/Join pattern
## Description

The Fork/Join pattern is a concurrency design approach that splits (“forks”) a large task into multiple smaller subtasks, processes these subtasks in parallel, and then recombines (“joins”) their results. In Java, this pattern is embodied by the `ForkJoinPool` and `RecursiveTask` / `RecursiveAction` classes within the `java.util.concurrent` package. It’s particularly useful for CPU-intensive, divide-and-conquer problems where parallel processing can significantly reduce completion time.

### Key Elements

- **Divide-and-Conquer**: Recursively split tasks into smaller chunks until reaching a manageable size.
- **Parallel Processing**: Distribute subtasks across a pool of worker threads (usually a `ForkJoinPool`).
- **Work Stealing**: Idle threads “steal” tasks from busy threads’ queues to optimize utilization.
- **RecursiveTask / RecursiveAction**: Abstract classes for defining the task logic and splitting strategy (`RecursiveTask` returns a result, `RecursiveAction` does not).

## References

1. [Java Documentation for `ForkJoinPool`](https://docs.oracle.com/en/java/javase/21//docs/api/java.base/java/util/concurrent/ForkJoinPool.html)  
2. [Java Concurrency in Practice](https://www.amazon.com/Java-Concurrency-Practice-Brian-Goetz/dp/0321349601) – Brian Goetz  
3. [Java Design Patterns – Contribution Guidelines](https://github.com/iluwatar/java-design-patterns/wiki)

## Acceptance Criteria

- [ ] Create a new module or package named `fork-join` (or similar).
- [ ] Demonstrate splitting a computationally intensive task into subtasks using `RecursiveTask` or `RecursiveAction`.
- [ ] Show how the results are joined together for a final outcome.
- [ ] Provide a README (or `.md` file) explaining the pattern with diagrams or code snippets.
- [ ] Include tests verifying parallel execution and correctness of results.
- [ ] Ensure all code meets the repository style, naming conventions, and passes CI checks.

### PR DESCRIPTION
**Pull request by SandhyaDevadiga**
Fixes #3227

## What this PR does

Implements the Fork/Join concurrency pattern using a parallel array
summation example.

## Files added

- `SumTask.java` — RecursiveTask that splits the array and sums in parallel
- `ForkJoinSumCalculator.java` — Wrapper class that manages the ForkJoinPool
- `App.java` — Demo showing parallel sum of 10 million numbers
- `SumTaskTest.java` — 6 unit tests for the recursive task
- `ForkJoinSumCalculatorTest.java` — 5 unit tests for the calculator
- `README.md` — Pattern explanation with examples and diagrams
- `pom.xml` — Maven module configuration

### CODE REVIEW

File: fork-join/src/main/java/com/iluwatar/forkjoin/SumTask.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+   * @return the sum of elements from start to end
+   */
+  @Override
+  protected Long compute() {
+    int length = end - start;
+
+    // BASE CASE: if the chunk is small enough, just sum directly
+    if (length <= THRESHOLD) {
+      return computeDirectly();
+    }
```
[the remark this answers is not in the document]
Comment (SandhyaDevadiga) [reply fork-join/src/main/java/com/iluwatar/forkjoin/SumTask.java:56]: Good catch. Added validation in the constructor to throw IllegalArgumentException when start > end. Also added a test for this edge case.

**Review by iluwatar — CHANGES_REQUESTED** [review]
You need to add the new module to the parent pom.xml, otherwise CI does not build it

**Comment by SandhyaDevadiga** [comment]
All checks passing. Ready for review.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
@all-contributors please add @SandhyaDevadiga for code
